controlplane/kubeadm/internal/cluster.go

```diff
@@ -28,22 +28,23 @@ import (
 	"k8s.io/apimachinery/pkg/runtime"
 	kerrors "k8s.io/apimachinery/pkg/util/errors"
 	"k8s.io/client-go/kubernetes/scheme"
-	"sigs.k8s.io/controller-runtime/pkg/client"
-	ctrlclient "sigs.k8s.io/controller-runtime/pkg/client"
-
 	clusterv1 "sigs.k8s.io/cluster-api/api/v1alpha3"
 	"sigs.k8s.io/cluster-api/controllers/remote"
 	"sigs.k8s.io/cluster-api/controlplane/kubeadm/internal/machinefilters"
 	"sigs.k8s.io/cluster-api/util/secret"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	ctrlclient "sigs.k8s.io/controller-runtime/pkg/client"
 )
 
 // ManagementCluster defines all behaviors necessary for something to function as a management cluster.
 type ManagementCluster interface {
 	ctrlclient.Reader
 
 	GetMachinesForCluster(ctx context.Context, cluster client.ObjectKey, filters ...machinefilters.Func) (FilterableMachineCollection, error)
-	TargetClusterEtcdIsHealthy(ctx context.Context, clusterKey client.ObjectKey) error
-	TargetClusterControlPlaneIsHealthy(ctx context.Context, clusterKey client.ObjectKey) error
+	TargetClusterControlPlaneHealthCheck(ctx context.Context, controlPlane *ControlPlane, clusterKey client.ObjectKey) (HealthCheckResult, error)
+	TargetClusterEtcdHealthCheck(ctx context.Context, controlPlane *ControlPlane, clusterKey client.ObjectKey) (HealthCheckResult, error)
+	TargetClusterAnalyseEtcdHealth(ctx context.Context, controlPlane *ControlPlane, clusterKey client.ObjectKey) error
+	TargetClusterAnalyseControlPlaneHealth(ctx context.Context, controlPlane *ControlPlane, clusterKey client.ObjectKey) error
```

Comment:
I'm not sure this fits into this PR, but I find the following points somehow confusing
- We are using the term `TargetCluster`, which is quite unusual for CAPI. Should we use `WorkloadCluster` instead?
- Why are those methods are defined on the `ManagementCluster` interface, if their goal is to access the status of elements (Etcd and ControlPlane) in a workload cluster. Should we move those methods to the `WorkloadCluster` interface instead, so we can definitely drop `Target/WorkloadCluster` the prefix?
- The difference between `HealthCheck` and `AnalyseEtcdHealth` is not really intuitive if I look at the func name only

Reply:
Moved all of them out of Management interface.